Src/SwqlStudio/Subscriptions/SubscriptionCallbacks.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace SwqlStudio.Subscriptions
+{
+    internal class SubscriptionCallbacks
+    {
+        private readonly string activeSubscriberAddress;
+        internal string Uri { get; }
+        internal string Id { get; set; }
+
+        private readonly NotificationDeliveryServiceProxy proxy;
+        private readonly object itemsLock = new object();
+
+        internal IEnumerable<SubscriberCallback> Callbacks
+        {
+            get
+            {
+                lock (this.itemsLock)
+                {
+                    return new List<SubscriberCallback>(this.callbacks);
+                }
+            }
+        }
+
+        public bool Empty
+        {
+            get
+            {
+                lock (this.itemsLock)
+                {
+                    return !this.callbacks.Any();
+                }
+            }
+        }
+
+        private readonly List<SubscriberCallback> callbacks = new List<SubscriberCallback>();
```

Comment:
Originally there was only one callback possible per registration. Now multiple tabs may register their callback to obtain notifications.